File: software/examples/java/ExampleMotor.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletMotorizedLinearPoti;

public class ExampleMotor {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Motorized Linear Poti Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletMotorizedLinearPoti mlp =
		  new BrickletMotorizedLinearPoti(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add position reached listener
		mlp.addPositionReachedListener(new BrickletMotorizedLinearPoti.PositionReachedListener() {
			public void positionReached(int position) {
				System.out.println("Position: " + position); // Range: 0 to 100
			}
		});

		// Move slider smooth to the middle
		mlp.setMotorPosition(50, BrickletMotorizedLinearPoti.DRIVE_MODE_SMOOTH, false);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/c/example_motor.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_motorized_linear_poti.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change XYZ to the UID of your Motorized Linear Poti Bricklet

// Callback function for position reached callback
void cb_position_reached(uint16_t position, void *user_data) {
	(void)user_data; // avoid unused parameter warning

	printf("Position: %u\n", position); // Range: 0 to 100
}

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	MotorizedLinearPoti mlp;
	motorized_linear_poti_create(&mlp, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Register position reached callback to function cb_position_reached
	motorized_linear_poti_register_callback(&mlp,
	                                        MOTORIZED_LINEAR_POTI_CALLBACK_POSITION_REACHED,
	                                        (void (*)(void))cb_position_reached,
	                                        NULL);

	// Move slider smooth to the middle
	motorized_linear_poti_set_motor_position(&mlp, 50,
	                                         MOTORIZED_LINEAR_POTI_DRIVE_MODE_SMOOTH,
	                                         false);

	printf("Press key to exit\n");
	getchar();
	motorized_linear_poti_destroy(&mlp);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}
